hack/chains/cosign-verify-tasks.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
+metadata:
+  annotations:
+    tekton.dev/displayName: cosign verify
+    tekton.dev/pipelines.minVersion: "0.19"
+    tekton.dev/tags: cosign, chains, signature
+  labels:
+    app.kubernetes.io/version: "0.1"
+    operator.tekton.dev/provider-type: redhat
+  name: cosign-verify
+spec:
+  description: cosign-verify verifies the signature of the image reference with the provided public key.
+  params:
+  - description: Public key used to verify the signature. Points to a Secret containing `cosign.pub` key.
+    name: PUBLIC_KEY_NAME
+    type: string
+    default: signing-secrets
+  - description: Pull reference of the image to verify.
+    name: IMAGE
+    type: string
+  results:
+    - name: VERIFY_JSON
+      description: Result of verification in JSON format
+  steps:
+  - script: |
+      #!/usr/bin/env bash
+      cosign verify --output-file $(results.VERIFY_JSON.path) --key <(echo "$PUBLIC_KEY") $(params.IMAGE)
```

Comment:
Since this is always running on k8s/ocp and the key will always come from a Secret, maybe we can use the syntax `--key k8s://<namespace>/<secret-name>`  ?

Reply:
I did have this in the initial version. The reason I did it this way is that the private key is not exposed to the cosign in this task. Perhaps that is taking two steps when we could do one. I wanted to be sure we can provide just the public key.

Reply:
Just realized that the task accesses the secret nonetheless, I'm getting old...